overcast/runner/__init__.py

```diff
@@ -128,6 +128,83 @@ def get_creds_from_env():
 #    d['cacert'] = os.environ.get('OS_CACERT', None)
     return d
 
+
+class node(object):
+    dr = self.DeploymentRunner()
+
+    def __init__(self, name, info, keypair, userdata, mappings=None, record_resource=None):
+        self.record_resource = lambda *args, **kwargs: None
+        self.info     = info
+        self.networks = dr.networks or {}
+        self.secgroups = dr.secgroups or {}
+        self.keypair  = keypair
+        self.userdata = userdata
+        self.image    = None
+        self.flavor   = None
+        self.mappings = mappings or {}
+        self.server_id = None
+        self.server_status = None
+
+    def poll(self, desired_status = 'ACTIVE'):
+        if self.server_status != desired_status:
+            nc = dr.get_nova_client()
+            self.server_status = nc.servers.get(self.server_id).status
+        return self.server_status
+
+    def clean(self):
+        nc = dr.get_nova_client()
+        server = nc.servers.get(self.server_id)
+        server.delete()
```

Comment:
You need to also delete the created port. You also need to either delete the floating ip or keep track of it so that you avoid creating an extra one in case of a rebuild.